contracts/lib/StateLib.sol

```diff
@@ -68,11 +68,13 @@ library StateLib {
         id1 => state2 => [index1]
          */
         mapping(uint256 => mapping(uint256 => uint256[])) stateIndexes;
+        // exisitingTypeIds is used to keep track of existing typeIds
+        mapping(bytes2 => bool) existingTypeIds;
```

Comment:
I would rename `existing` to `supported`, and function isIdTypeSupported. Because type ids always exist